src/main/java/com/google/sps/servlets/BusinessDataServlet.java

```diff
@@ -0,0 +1,207 @@
+      List<String> reviews = reviewsArr == null ? new ArrayList<String>() : Arrays.asList(reviewsArr);
+
+  
+      Business business = new Business(name, categories, rating, addressLat, addressLng, address, logoUrl, picturesUrls, desc, menuLink, orderDetails, contactDetails, businessLink);
+      businesses.add(business);
+    }
+
+    // Send the JSON as the response.
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(businesses));
+  }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    // Get the input from the form.
+    String name = request.getParameter(Constants.BUSINESS_NAME);
+    String desc = request.getParameter(Constants.BUSINESS_DESC);
+    String[] categoriesArr = request.getParameterValues(Constants.BUSINESS_CATEGORIES);
+    List<String> categories = categoriesArr == null ? new ArrayList<String>() : Arrays.asList(categoriesArr);
+    String address = request.getParameter(Constants.BUSINESS_ADDRESS);
+    String addressLat = request.getParameter(Constants.BUSINESS_ADDRESS_LAT);
+    String addressLng = request.getParameter(Constants.BUSINESS_ADDRESS_LNG);
+    String contactDetails = request.getParameter(Constants.BUSINESS_CONTACT_INFO);
+    String orderDetails = request.getParameter(Constants.BUSINESS_ORDER_INFO);
+    String businessLink = request.getParameter(Constants.BUSINESS_LINK);
+    String menuLink = request.getParameter(Constants.BUSINESS_MENU_LINK);
+    String logoUrl = getUploadedLogoUrlFromBlobstore(request, Constants.BUSINESS_LOGO);
+    List<String> picturesUrls = getUploadedPicturesUrlsFromBlobstore(request, Constants.BUSINESS_PICTURES);
+    // can't add reviews and rating when creating a new business
+    String rating = "";
+    List<String> reviews = new ArrayList<String>();
+
+    Entity businessEntity = new Entity("Business");
+    businessEntity.setProperty(Constants.BUSINESS_NAME, name);
+    businessEntity.setProperty(Constants.BUSINESS_DESC, desc);
+    businessEntity.setProperty(Constants.BUSINESS_CATEGORIES, gson.toJson(categories));
+    businessEntity.setProperty(Constants.BUSINESS_ADDRESS, address);
+    businessEntity.setProperty(Constants.BUSINESS_ADDRESS_LAT, addressLat);
+    businessEntity.setProperty(Constants.BUSINESS_ADDRESS_LNG, addressLng);
+    businessEntity.setProperty(Constants.BUSINESS_CONTACT_INFO, contactDetails);
+    businessEntity.setProperty(Constants.BUSINESS_ORDER_INFO, orderDetails);
+    businessEntity.setProperty(Constants.BUSINESS_LINK, businessLink);
+    businessEntity.setProperty(Constants.BUSINESS_MENU_LINK, menuLink);
+    businessEntity.setProperty(Constants.BUSINESS_LOGO, logoUrl);
+    businessEntity.setProperty(Constants.BUSINESS_PICTURES, gson.toJson(picturesUrls));
+    businessEntity.setProperty(Constants.BUSINESS_RATING, rating);
+    businessEntity.setProperty(Constants.BUSINESS_REVIEWS, reviews);
```

Comment:
Consider using a static import (http://www.javapractices.com/topic/TopicAction.do?Id=195) so that you can leave out the Constants. prefix? I think it can be justified in this case since the var names aren't likely to be repeated and there are way too many repeats of "Constants."

Reply:
oh that's cool. thanks! I've done this!